### DISCUSSION TITLE
QUESTION｜关于将 Action 自动构建的内容发行到 Release 中是否可取

### DISCUSSION THREAD

**Discussion by YaoSiQian** [MCSLTeam/MCSL2#82] [提问区域]
- [x] 未在 [Issues](https://github.com/LxHTT/MCSL2/issues) 中看到其他相同疑问
___

### 客观情况
目前，MCSL 2 使用了 Github Actions 来承担自动构建的内容。根据 [build.yml](https://github.com/LxHTT/MCSL2/blob/master/.github/workflows/build.yml#L3)，每一次进行`push`操作后，都会构建适用于 Windows、macOS 以及 Linux 系统的可执行文件，并且**将其以*正式发行版*的方式发布到 Release**。

### *可能*发生的状况
萌新盲目按照[最新版](https://github.com/LxHTT/MCSL2/releases/latest)进行下载，结果遇到类似于 #22 的状况 ~~，然后大骂傻逼软件~~。

### 个人观点
反对将最新构建发布到 Release 区域，而是通过 Actions 下载工件。

### 观点原因
- 发行版通常都是经过了QA的，不能莽然发布
- 无法保证每一个 Commit 的质量

### 中和方法
将最新构建发布为 Pre-Release，比如 #23

**Comment by lxhtt** [comment]
讲个鬼故事（
更新服务并不会和GitHub相通（

**Comment by Qubik65536** [comment]
将不稳定构建发布到 Release 并非不可取，甚至很多著名开源项目都是这么做的，例如 [NeoVim](https://github.com/neovim/neovim/releases/tag/nightly) 和 [Adoptium JDK](https://github.com/adoptium/temurin8-binaries/releases/tag/jdk8u352-b05-ea) 都会将 Nighty 构建发布，**但将 beta 标记为正式 latest 发布不是一个好习惯**。尤其当你希望用户从 GitHub Release 页下载软件的时候（你是这么做的，官网下载最先看到的链接就指向 GitHub），这种行为更不可推荐。

**Comment by lxhtt** [comment]
将会在每次push之前决定是否要release或者pre-release。
